### ISSUE TITLE
Remove 6 second delay from login screen(s)

### ISSUE THREAD

**Issue by blobaugh** [WebDevStudios/aad-first-party-sso-wordpress#4]
The login screen takes a significant amount of time to load. This is due to it making a request to login.windows.net whenever the page loads. That request is taking **6 seconds** to perform. It is not strictly necessary.

Here is the function that is running: https://github.com/WebDevStudios/aad-first-party-sso-wordpress/blob/master/Settings.php#L144

It should be updated to cache the json output in a transient for a while and look there. It currently attempts to load a file. Maybe that was a caching attempt but it will never work.

Should be an easy refactor.

**Comment by jtsternberg** [comment]
> It currently attempts to load a file. Maybe that was a caching attempt but it will never work.

Just a clarification, that json file location was what the original author was using [to configure the plugin](https://github.com/psignoret/aad-sso-wordpress/tree/47c17db4c2cb85d602ac972a83b3f2fc8fbdce8e#3-configure-the-plugin) (because there was no admin settings).

**Comment by blobaugh** [comment]
Right, it is still going out attempting to locate a json file. Is that to set setting defaults or should it be removed in this version?

**Comment by jtsternberg** [comment]
I mean, it could theoretically still be used to load settings (instead of using the DB), so whether that should be removed is up to you, I guess. The idea, though, about caching the lookup to login.windows.net makes perfect sense to me.

**Comment by blobaugh** [comment]
We want to have the settings in the db though ;). If this json file is not loading defaults in that we need to care about then I vote we remove it and save ourselves the 6 seconds of load time.
